raft/src/test/java/org/apache/kafka/raft/LeaderStateTest.java

```diff
@@ -36,30 +36,31 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class LeaderStateTest {
+public class LeaderStateTest<T> {
```

Comment:
I confess it's a little annoying to see the generic type leak down to here, but I guess that's the price we have to pay. There's a big part of me that wants to remove the generic and let `ApiMessageAndVersion` be the only supported type. Anyway, that is fuel for a separate issue/discussion.

Reply:
Why is the `T` leaked to the tests? Glancing at the code the type parameter `T` is never used. Did you try changing the signature to:

```
private final BatchAccumulator<?> accumulator = Mockito.mock(BatchAccumulator.class);
```